src/protocol/vm/state.rs

```diff
@@ -541,6 +544,111 @@ impl VMPoolState<PreCachedDB> {
 
         merged
     }
+
+    async fn get_amount_out(
+        &self,
+        sell_token: H160,
+        sell_amount: U256,
+        buy_token: H160,
+    ) -> Result<GetAmountOutResult, SimulationError> {
+        let mut sell_amount_respecting_limit = sell_amount;
+        let mut sell_amount_exceeds_limit = false;
+        let overwrites = self
+            .get_overwrites(
+                vec![sell_token, buy_token],
+                U256::from_big_endian(&(*MAX_BALANCE / rU256::from(100)).to_be_bytes::<32>()),
+            )
+            .await?;
+
+        let sell_amount_limit = self
+            .clone()
+            .get_sell_amount_limit(vec![sell_token, buy_token], Some(overwrites.clone()))
+            .await?;
+
+        if self
+            .capabilities
+            .contains(&Capability::HardLimits) &&
+            sell_amount_limit < sell_amount
+        {
+            sell_amount_respecting_limit = sell_amount_limit;
+            sell_amount_exceeds_limit = true;
+        }
+
+        let overwrites_with_sell_limit = self
+            .clone()
+            .get_overwrites(vec![sell_token, buy_token], sell_amount_limit)
+            .await?;
+        let complete_overwrites = self.merge(&overwrites, &overwrites_with_sell_limit);
+        let pool_id = self.clone().id;
```

Comment:
Is it possible to clone only the id instead of the whole pool?
```suggestion
        let pool_id = self.id.clone();
```